SimGeneral/CaloAnalysis/plugins/CaloTruthAccumulator.cc

```diff
@@ -426,13 +430,7 @@ template<class T> void CaloTruthAccumulator::fillSimHits( std::vector<std::pair<
       DetId id(0);
       const uint32_t simId = simHit.id();
       if( isHcal ) {
-	int subdet, z, depth0, eta0, phi0, lay;
-	HcalTestNumbering::unpackHcalIndex(simId, subdet, z, depth0, eta0, phi0, lay);
-	int sign = (z==0) ? (-1):(1);
-	HcalDDDRecConstants::HcalID tempid = hcddd_->getHCID(subdet, sign*eta0, phi0, lay, depth0);
-	if (subdet==int(HcalEndcap)) {
-	  id = HcalDetId(HcalEndcap,sign*tempid.eta,tempid.phi,tempid.depth);    
```

Comment:
Out of curiosity: the fact that before "id" was assigned only for HcalEndcap (while now it is for any Hcal) was a kind of bug also fixed with this PR?